JinGine.WinForms/Program.cs

```diff
@@ -25,17 +27,19 @@ public static void Main()
 
         var container = new UnityContainer()
             .RegisterInstance(new AppSettings(Settings.Default.FilesPath), InstanceLifetime.Singleton)
-            .RegisterInstance(EventAggregator.Instance, InstanceLifetime.Singleton)
-            .RegisterType<IFileManager, FileManagerFacade>()
-            .RegisterType<ICommandHandler<OpenBinaryFileCommand>, OpenBinaryFileCommandHandler>()
-            .RegisterType<ICommandHandler<OpenCSharpFileCommand>, OpenCSharpFileCommandHandler>()
-            .RegisterType<ICommandDispatcher, CommandDispatcher>()
+            .RegisterInstance<IEventAggregator>(EventAggregator.Instance, InstanceLifetime.Singleton)
+            .RegisterSingleton<IFileManager, FileManagerFacade>()
+            .RegisterSingleton<ICommandHandler<OpenBinaryFileCommand>, OpenBinaryFileCommandHandler>()
+            .RegisterSingleton<ICommandHandler<OpenCSharpFileCommand>, OpenCSharpFileCommandHandler>()
+            .RegisterSingleton<ICommandDispatcher, CommandDispatcher>()
             .RegisterSingleton<MainMenuFactory>()
+            .RegisterSingleton<IBinaryFileSerializer, BinaryFileSerializer>()
             .AddExtension(new Diagnostic());
```

Comment:
doesn't need to be done immediatelly, but should be extracted to a reusable custom `UnityContainerRegistrar`